r/src/r_to_arrow.cpp

```diff
@@ -1089,6 +1096,7 @@ class RStructConverter : public StructConverter<RConverter, RConverterTrait> {
       auto fields = this->struct_type_->fields();
       R_xlen_t n_columns = XLENGTH(values);
 
+      // TODO: this needs to be updated for the new struct situation like Extend above
```

Comment:
I haven't touched the `DelayedExtend` definition at all yet. I need to propagate the changes made in `Extend` here, and will once that approach is more solidified